Modules/Filtering/DistanceMap/include/itkDanielssonDistanceMapImageFilter.hxx

```diff
@@ -30,16 +30,14 @@ namespace itk
 template <typename TInputImage, typename TOutputImage, typename TVoronoiImage>
 DanielssonDistanceMapImageFilter<TInputImage, TOutputImage, TVoronoiImage>::DanielssonDistanceMapImageFilter()
 {
-  this->SetNumberOfRequiredOutputs(3);
+  constexpr unsigned int numberOfRequiredOutputs{ 3 };
+  this->SetNumberOfRequiredOutputs(numberOfRequiredOutputs);
 
-  // distance map
-  this->SetNthOutput(0, this->MakeOutput(0));
-
-  // voronoi map
-  this->SetNthOutput(1, this->MakeOutput(1));
-
-  // distance vectors
-  this->SetNthOutput(2, this->MakeOutput(2));
+  // distance map, voronoi map, distance vectors
+  for (unsigned int i{}; i < numberOfRequiredOutputs; ++i)
+  {
+    ProcessObject::SetNthOutput(i, Self::MakeOutput(i));
+  }
```

Comment:
Maybe we can make this more generic, like:

    ProcessObject::SetRequiredOutputs(*this, numberOfRequiredOutputs);

This new function, `SetRequiredOutputs` would then set the numberOfRequiredOutputs _and_ do the `for` loop. What do you think?

Reply:
For the constructors we are in a bit of a hard place where the methods called be the constructor should not be virtual. That is the virtual MakeOutput should not be called by base classes in the constructor. 

This is more of a design issue, maybe the output should only be created on demand? or Maybe a separate "virtual Initialize" method is needed.

Then there is the idea of having a base class take multiple template parameters for inputs and outputs.... But for the few filters it's likely not worth the refactor.